src/coreclr/pal/src/arch/riscv64/context2.S

```diff
@@ -9,4 +9,238 @@
 #include "unixasmmacros.inc"
 #include "asmconstants.h"
 
-#error "TODO-RISCV64: missing implementation"
+// Incoming:
+//  a0: Context*
+//  a1: Exception*
+//
+LEAF_ENTRY RtlRestoreContext, _TEXT
+#ifdef HAS_ASAN
+// TODO-RISCV64: missing implementation
+#endif
+
+    addi t4, a0, 0
+    lw  t1, CONTEXT_ContextFlags(t4)
+    andi t1, t1, 0x1 << CONTEXT_FLOATING_POINT_BIT
+    beq t1, zero, LOCAL_LABEL(No_Restore_CONTEXT_FLOATING_POINT)
+
+    //64-bits FPR.
+    addi t0, t4, CONTEXT_FPU_OFFSET
+
+    fld  f0, (CONTEXT_F0)(t0)
+    fld  f1, (CONTEXT_F1)(t0)
+    fld  f2, (CONTEXT_F2)(t0)
+    fld  f3, (CONTEXT_F3)(t0)
+    fld  f4, (CONTEXT_F4)(t0)
+    fld  f5, (CONTEXT_F5)(t0)
+    fld  f6, (CONTEXT_F6)(t0)
+    fld  f7, (CONTEXT_F7)(t0)
+    fld  f8, (CONTEXT_F8)(t0)
+    fld  f9, (CONTEXT_F9)(t0)
+    fld  f10, (CONTEXT_F10)(t0)
+    fld  f11, (CONTEXT_F11)(t0)
+    fld  f12, (CONTEXT_F12)(t0)
+    fld  f13, (CONTEXT_F13)(t0)
+    fld  f14, (CONTEXT_F14)(t0)
+    fld  f15, (CONTEXT_F15)(t0)
+    fld  f16, (CONTEXT_F16)(t0)
+    fld  f17, (CONTEXT_F17)(t0)
+    fld  f18, (CONTEXT_F18)(t0)
+    fld  f19, (CONTEXT_F19)(t0)
+    fld  f20, (CONTEXT_F20)(t0)
+    fld  f21, (CONTEXT_F21)(t0)
+    fld  f22, (CONTEXT_F22)(t0)
+    fld  f23, (CONTEXT_F23)(t0)
+    fld  f24, (CONTEXT_F24)(t0)
+    fld  f25, (CONTEXT_F25)(t0)
+    fld  f26, (CONTEXT_F26)(t0)
+    fld  f27, (CONTEXT_F27)(t0)
+    fld  f28, (CONTEXT_F28)(t0)
+    fld  f29, (CONTEXT_F29)(t0)
+    fld  f30, (CONTEXT_F30)(t0)
+    fld  f31, (CONTEXT_F31)(t0)
+
+    lw t1, (CONTEXT_FLOAT_CONTROL_OFFSET)(t0)
+    fscsr x0, t1
+
+LOCAL_LABEL(No_Restore_CONTEXT_FLOATING_POINT):
+
+    lw  t1, CONTEXT_ContextFlags(t4)
+    andi t1, t1, 0x1 << CONTEXT_INTEGER_BIT
+    beq t1, zero, LOCAL_LABEL(No_Restore_CONTEXT_INTEGER)
```

Comment:
can be simplified with `beqz t1, LOCAL_LABEL(No_Restore_CONTEXT_INTEGER)` alias

Reply:
Fixed